File: bootcamp-2023/integration-testing/exercise/php/src/TripServiceKata/Trip/TripDAO.php
<?php

namespace TripServiceKata\Trip;

use TripServiceKata\User\User;
use TripServiceKata\Exception\DependentClassCalledDuringUnitTestException;

class TripDAO
{
    public static function findTripsByUser(User $user)
    {
        return $user->getTrips();
    }
}

File: bootcamp-2023/integration-testing/exercise/kotlin/src/main/kotlin/org/craftedsw/tripservicekata/trip/TripDAO.kt
package org.craftedsw.tripservicekata.trip

import org.craftedsw.tripservicekata.exception.CollaboratorCallException
import org.craftedsw.tripservicekata.user.User

class TripDAO {

    companion object {
        @JvmStatic fun findTripsByUser(user: User): List<Trip> {
            return user.trips
        }
    }

}
